owncloudApp/src/androidTest/java/com/owncloud/android/shares/db/OCShareDaoTest.kt

```diff
@@ -167,183 +163,143 @@ class OCShareDaoTest {
+        assertEquals("Document 1 link", document1SharesForUser3.get(0).name)
+        assertEquals("http://server:port/s/3", document1SharesForUser3.get(0).shareLink)
     }
 
     @Test
     fun getNonExistingPublicShare() {
-        ocShareDao.insert(
-            listOf(
-                TestUtil.createPublicShare(
-                    path = "/Videos/video1.mp4",
-                    isFolder = false,
-                    accountOwner = "user@server",
-                    name = "Video 1 link",
-                    shareLink = "http://server:port/s/1"
-                )
-            )
-        )
+        ocShareDao.insert(listOf(newPublicShare()))
 
-        val video2Shares = getValue(
+        val nonExistingShare = getValue(
             ocShareDao.getSharesForFileAsLiveData(
-                "/Videos/video2.mp4", "user@server", listOf(ShareType.PUBLIC_LINK.value)
+                "/Texts/text2.txt", "user@server", listOf(ShareType.PUBLIC_LINK.value)
             )
         )
-
-        assertThat(video2Shares, notNullValue())
-        assertEquals(video2Shares.size, 0)
+        assertThat(nonExistingShare, notNullValue())
+        assertEquals(0, nonExistingShare.size)
     }
 
     @Test
     fun replacePublicShareIfAlreadyExists_exists() {
-        ocShareDao.insert(
-            listOf(
-                TestUtil.createPublicShare(
-                    path = "/Texts/text1.txt",
-                    isFolder = false,
-                    name = "Text 1 link",
-                    shareLink = "http://server:port/s/1"
-                )
-            )
-        )
+        ocShareDao.insert(listOf(newPublicShare()))
 
         ocShareDao.replaceSharesForFile(
-            listOf( // Update link name
-                TestUtil.createPublicShare(
-                    path = "/Texts/text1.txt",
-                    isFolder = false,
-                    name = "Text 2 link",
-                    shareLink = "http://server:port/s/1"
-                )
-            )
+            listOf(newPublicShare(name = "Text 2 link"))
         )
 
         val textShares = getValue(
             ocShareDao.getSharesForFileAsLiveData(
                 "/Texts/text1.txt", "admin@server", listOf(ShareType.PUBLIC_LINK.value)
             )
         )
-
         assertThat(textShares, notNullValue())
-        assertEquals(textShares.size, 1)
-        assertEquals(textShares.get(0).name, "Text 2 link")
+        assertEquals(1, textShares.size)
+        assertEquals("Text 2 link", textShares.get(0).name)
     }
 
     @Test
     fun replacePublicShareIfAlreadyExists_doesNotExist() {
-        ocShareDao.insert(
-            listOf(
-                TestUtil.createPublicShare(
-                    path = "/Texts/text1.txt",
-                    isFolder = false,
-                    name = "Text 1 link",
-                    shareLink = "http://server:port/s/1"
-                )
-            )
-        )
+        ocShareDao.insert(listOf(newPublicShare()))
```

Comment:
Haven't you created a method for a just one item instead of a list?

Reply:
Done